src/main/java/seedu/address/model/person/DeliveryDate.java

```diff
@@ -0,0 +1,82 @@
+package seedu.address.model.person;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+
+
+/**
+ * Represents an Order's delivery datre in the CakeCollate.
+ * Guarantees: immutable; is valid as declared in {@link #isValidDeliveryDate(String)}
+ */
+public class DeliveryDate {
```

Comment:
Looks well-implemented to me, seems pretty extensible if we need to add more formats or features